### PR TITLE
[Documentation] Set error placement in contract, match example to specified order

### LINKED ISSUE #12614: [Docs] Order of contract parts / Solidity Style Guide maintenance
## Page

https://docs.soliditylang.org/en/latest/style-guide.html?highlight=style%20guide

## Abstract

It is specified:

> Inside each contract, library or interface, use the following order:

But this advice is not followed in examples.

Screen Shot 2022-02-01 at 19 14 07
Screen Shot 2022-02-01 at 19 14 25

THE QUESTION:

> Should constructor be specified to be above modifiers?
> Or should constructor be moved down with the functions in that example?

## Pull request

(I can provide after answering THE QUESTION)

### PR DESCRIPTION
**Pull request by fulldecent**
closes #12614

### CODE REVIEW

File: docs/style-guide.rst [hunk 1]
```diff
@@ -1058,8 +1058,9 @@ Inside each contract, library or interface, use the following order:
 1. Type declarations
 2. State variables
 3. Events
-4. Modifiers
-5. Functions
+4. Errors
```
Comment (nikola-matic) [inline docs/style-guide.rst:1061]: Do we have any examples of errors in the style guide? If not, I would move this change to another issue/PR pair where the `4. Error` entry can be added, with some examples.

**Review by nikola-matic — CHANGES_REQUESTED** [review]
There's another miss-ordered `constructor()` definition in the same section, on line 1134 that should go after the modifier.

File: docs/style-guide.rst (hunk 1 above)
Comment (fulldecent) [reply docs/style-guide.rst:1061]: Added example at f18f4f06d4f94cb680f2ad661f872baa6fc9afd8

**Comment by fulldecent** [comment]
@nikola-matic Cool, other fixed at eb68c688910972b5574eaaeb567a00421a69fc6c

File: docs/style-guide.rst [hunk 2]
```diff
@@ -1058,14 +1058,40 @@ Inside each contract, library or interface, use the following order:
...
+6. Functions
 
 .. note::
 
     It might be clearer to declare types close to their use in events or state
     variables.
 
+Yes:
+
+.. code-block:: solidity
```
Comment (nikola-matic) [inline docs/style-guide.rst:1072]: So, while this is a good start, as you've put examples into the 'Order of Layout' section, these feel somewhat lacking, as they only show the order of functions and errors, whereas the section describes everything from type declarations, state variables, functions, events, etc.
On the other hand, I'm also not sure how useful the `No` section would be here, as you'd have quite a few permutations of the wrong way to do it, so I think that just having the `Yes` sections (without the actual Yes label) would make more sense. As in:
```
pragma version;

contract C {
  type declarations;
  state variables;
  events;
  errors:
  modifiers;
  functions;
}
```

Comment (fulldecent) [reply docs/style-guide.rst:1072]: This is not a documentation rewrite (yet).

I'm hoping this PR can be judged on delta terms not absolute terms.

Comment (nikola-matic) [reply docs/style-guide.rst:1072]: That's fair - but do you have plans for extending this section later, as it kinda feels incomplete to just have a contract with an error and a function?

**Comment by nikola-matic** [comment]
@fulldecent can you try and push an empty commit + rebase (squash) all of your commits into one to hopefully get the CI pipeline running again?

File: docs/style-guide.rst [hunk 3]
```diff
@@ -1058,14 +1058,40 @@ Inside each contract, library or interface, use the following order:
...
+        error DivideByZero();
+        function divide(int256 numerator, int256 denominator) public virtual returns (uint256);
+    }
+
+No:
+
+.. code-block:: solidity
+
+    // SPDX-License-Identifier: GPL-3.0
+    pragma solidity >=0.8.0 <0.9.0;
```
Comment (nikola-matic) [inline docs/style-guide.rst]: By the way, custom errors were only introduced in `0.8.4`. I'll push the required changes.

**Review by nikola-matic — APPROVED** [review]

**Pull request merged**